### PR TITLE
Improve hacking doc and pr template

### PR DESCRIPTION
**Pull request by huwshimi**
## Done
- Updated the instructions for setting up and developing maas-ui.
- Improved the PR template instructions.

## Fixes
Fixes https://github.com/canonical-web-and-design/MAAS-squad/issues/1991.

### CODE REVIEW

**Comment by huwshimi** [comment]
@anthonydillon Hey, has someone that's done a few reviews for maas-ui it'd be good to know if you have any comments on these changes.

**Comment by anthonydillon** [comment]
I'll leave if @squidsoup is QA'ing

File: HACKING.md [hunk 1]
```diff
@@ -1,106 +1,364 @@
...
+Check what resources your computer has and then run:
+
+```shell
+multipass launch -c [the number of cores] -d [some amount of disk space] -m [the amount of ram] --name [the instance name]
+```
+
+You should end up with a command something like this:
+
+```shell
+multipass launch -c 4 -d 20G -m 16G --name dev
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply HACKING.md:307]: :) I think it's pretty much impossible to work on MAAS without it... but it cost a pretty penny.

**Comment by huwshimi** [comment]
> This is a big improvement. Very mac focused but is useful for people using Multipass.

@Caleb-Ellis you work in Ubuntu don't you? Is there anything missing or wrong for your setup?

File: HACKING.md [hunk 2]
```diff
@@ -1,106 +1,364 @@
...
+You should now be able to run maas-ui and log into your MAAS:
+
+```shell
+dotrun
 ```
 
+Once everything has built you can access the site using the hostname:
 
-### Install Docker inside your container (LXD and multipass)
+[http://dev.local:8400/MAAS/](http://dev.local:8400/MAAS/).
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply HACKING.md:80]: Ah right, I think we lost that with the single-spa migration maybe. I've filed https://github.com/canonical-web-and-design/maas-ui/issues/1265.

**Comment by Caleb-Ellis** [comment]
> > This is a big improvement. Very mac focused but is useful for people using Multipass.
> 
> @Caleb-Ellis you work in Ubuntu don't you? Is there anything missing or wrong for your setup?

@huwshimi Yeah I use LXD containers for setting up local MAASes, but I can add that as a follow-up.

**Comment by huwshimi** [comment]
@Caleb-Ellis thanks that would be great!

**Pull request merged**